bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -142,6 +148,18 @@ public void scheduledInitialize() {
                 return;
             }
 
+            if (logger.isTraceEnabled()) {
+                if (!(getApiDef())) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unable to getApiDef from Hayward's server");
+                    clearPolling(pollTelemetryFuture);
+                    clearPolling(pollAlarmsFuture);
+                    commFailureCount = 50;
+                    initPolling(60);
+                    return;
+                }
+            }
```

Comment:
What's the purpose of this?

Reply:
The getApiDef provides additional information in troubleshooting issues and collecting other user's configuration information to help reverse engineer the Hayward API.  It is not needed during normal runtime.

Reply:
Dead code which is only used by the developer should be avoided in the source folder. You could remove it and stash the changes to save it for further usage.

Reply:
It is not only used by the developer.  I don't have every water feature, pump, spa, spillover, sensor, light, etc. on my pool that Hayward offers.  The data this produces from other users having issues with the binding is helpful in reverse engineering the protocol.